vendor_hex_core.sh

```diff
@@ -6,61 +6,13 @@ if [[ -z "$1" ]]; then
   exit 1
 fi
 
-source_dir=$1/src
```

Comment:
This will likely clash with the re-vendoring PR

Reply:
By "clash" do you mean "create a merge conflict"? I don't mind rebasing and fixing it once #2421 gets merged.

Reply:
I've re-vendored `hex_core` (at 0.7.1) and force-pushed, to resolve conflicts with #2421.